src/mime.tsx

```diff
@@ -0,0 +1,184 @@
+import { IRenderMime } from '@jupyterlab/rendermime-interfaces';
+// import { mystIcon } from './icon';
+import { markdownParse } from './myst';
+import {
+  DOITransformer,
+  enumerateTargetsPlugin,
+  footnotesPlugin,
+  getFrontmatter,
+  GithubTransformer,
+  keysPlugin,
+  linksPlugin,
+  mathPlugin,
+  ReferenceState,
+  resolveReferencesPlugin,
+  RRIDTransformer,
+  WikiTransformer
+} from 'myst-transforms';
+import { VFile } from 'vfile';
+import { PageFrontmatter, validatePageFrontmatter } from 'myst-frontmatter';
+import { internalLinksPlugin, linkFactory } from './links';
+import { FrontmatterBlock } from '@myst-theme/frontmatter';
+import {
+  ReferencesProvider,
+  TabStateProvider,
+  Theme,
+  ThemeProvider
+} from '@myst-theme/providers';
+import { useParse } from 'myst-to-react';
+import { renderers } from './renderers';
+import { References } from 'myst-common';
+import { ReactWidget, UseSignal } from '@jupyterlab/apputils';
+import { unified } from 'unified';
+import { Signal } from '@lumino/signaling';
+import React from 'react';
+
+/**
+ * The MIME type for Markdown.
+ */
+export const MIME_TYPE = 'text/markdown';
+
+type MySTState = {
+  references: References;
+  frontmatter: PageFrontmatter;
+  mdast: any;
+};
+
+/**
+ * A mime renderer for displaying Markdown with embedded latex.
+ */
+export class RenderedMySTMarkdown
+  extends ReactWidget
+  implements IRenderMime.IRenderer
+{
+  /**
+   * Construct a new rendered markdown widget.
+   *
+   * @param options - The options for initializing the widget.
+   */
+  constructor(options: IRenderMime.IRendererOptions) {
+    super();
+    this._state = null;
+    this.resolver = options.resolver;
+    this.linkHandler = options.linkHandler;
+    this.node.dataset['mimeType'] = MIME_TYPE;
+    this.addClass('myst-RenderedMySTMarkdown');
```

Comment:
There is some css that requires this to be wrapped in a top-level `myst`, we can change the name of that, but it will need it to be the same as the cell, or added in.